doc: add unref for input in readline (#38019)
nodejs/node: doc/api/readline.md

a `'resize'` event on the `output` if or when the columns ever change
([`process.stdout`][] does this automatically when it is a TTY).

> When creating a `readline.Interface` using `stdin` as input stream, the program

Reviewer: ```suggestion
When creating a `readline.Interface` using `stdin` as input, the program
```

Author: Sorry, I think I confused myself there 😓 fixing...

will not terminate until it receives `EOF` (<kbd>Ctrl</kbd>+<kbd>D</kbd> on
Linux/macOS, <kbd>Ctrl</kbd>+<kbd>Z</kbd> followed by <kbd>Return</kbd> on
Windows).
If you want your application to exit without waiting for user input, you can

> `unref` the standard input stream:

Reviewer: Thinking about this a bit more, it's probably worth adding a link for folks who want to dig deeper:
```suggestion
[`unref`][] the standard input stream:
```

And add a reference at the end of the file:

```markdown
[`unref`]: net.md#net_socket_unref
```

a `'resize'` event on the `output` if or when the columns ever change
([`process.stdout`][] does this automatically when it is a TTY).

> In order to end the wait for the `stdin` input stream (so that the program
> does not appear to have hanged), `unref` function can be used:

Reviewer: I think it would be helpful to indicate why does the program hangs. What do you think of this:
```suggestion
When creating an `Interface` using `stdin` as input stream, the program will not
terminate until it receives `EOF` (<kbd>Ctrl</kbd>+<kbd>D</kbd>). If you want
your application to exit without waiting for user input, you can `unref` the
standard input:
```

Author: I like your suggestion, thanks!

a `'resize'` event on the `output` if or when the columns ever change
([`process.stdout`][] does this automatically when it is a TTY).

When creating a `readline.Interface` using `stdin` as input stream, the program

> will notterminate until it receives `EOF` (<kbd>Ctrl</kbd>+<kbd>D</kbd>).

Reviewer: ```suggestion
will not terminate until it receives `EOF` (<kbd>Ctrl</kbd>+<kbd>D</kbd>).
```

> will notterminate until it receives `EOF` (<kbd>Ctrl</kbd>+<kbd>D</kbd>).

Reviewer: Interestingly, it seems the manipulation to get `EOF` is different on Windows: https://www.devdungeon.com/content/windows-eof-ctrl-d-equivalent

> <kbd>CTRL</kbd>-<kbd>Z</kbd> followed by <kbd>return</kdb>

Maybe worth adding the info in the parentheses.

If you want your application to exit without waiting for user input, you can

> `unref` the standard input:

Reviewer: ```suggestion
`unref` the standard input stream:
```

Reviewer: I think you forgot this one ;)

Author: oops, my bad, fixing

a `'resize'` event on the `output` if or when the columns ever change
([`process.stdout`][] does this automatically when it is a TTY).

When creating a `readline.Interface` using `stdin` as input, the program
will not terminate until it receives `EOF` (<kbd>Ctrl</kbd>+<kbd>D</kbd> on

> Linux/MacOS, <kbd>Ctrl</kbd>+<kbd>Z</kbd> followed by <kbd>return</kbd> on

Reviewer: nit
```suggestion
Linux/macOS, <kbd>Ctrl</kbd>+<kbd>Z</kbd> followed by <kbd>Return</kbd> on
```
